api/v1alpha1/clusterissuer_types.go

```diff
@@ -17,22 +17,40 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"github.com/cert-manager/issuer-lib/api/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
-//+kubebuilder:resource:scope=Cluster
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].status"
+// +kubebuilder:printcolumn:name="Reason",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].reason"
+// +kubebuilder:printcolumn:name="Message",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].message"
+// +kubebuilder:printcolumn:name="LastTransition",type="string",type="date",JSONPath=".status.conditions[?(@.type==\"Ready\")].lastTransitionTime"
+// +kubebuilder:printcolumn:name="ObservedGeneration",type="integer",JSONPath=".status.conditions[?(@.type==\"Ready\")].observedGeneration"
+// +kubebuilder:printcolumn:name="Generation",type="integer",JSONPath=".metadata.generation"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
 
 // ClusterIssuer is the Schema for the clusterissuers API
 type ClusterIssuer struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   IssuerSpec   `json:"spec,omitempty"`
-	Status IssuerStatus `json:"status,omitempty"`
+	Spec   IssuerSpec            `json:"spec,omitempty"`
+	Status v1alpha1.IssuerStatus `json:"status,omitempty"`
+}
+
+func (vi *ClusterIssuer) GetStatus() *v1alpha1.IssuerStatus {
+	return &vi.Status
 }
 
+func (vi *ClusterIssuer) GetIssuerTypeIdentifier() string {
+	return "clusterissuers.sample-issuer.example.com"
+}
```

Comment:
suggestion: I think this function would benefit from comments encouraging users to change this value, right? This should be changed for every issuer based on this repo